test/modules/home/home_interactor_test.dart

```diff
@@ -0,0 +1,159 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+import 'package:quick_test/quick_test.dart';
+import 'package:survey/models/survey_info.dart';
+import 'package:survey/models/user_info.dart';
+import 'package:survey/modules/home/home_module.dart';
+import 'package:survey/repositories/auth_repository.dart';
+import 'package:survey/repositories/survey_repository.dart';
+import 'package:survey/services/locator/locator_service.dart';
+import 'package:tuple/tuple.dart';
+import '../../helpers/behavior_subject_generator.dart';
+import 'home_interactor_test.mocks.dart';
+
+@GenerateMocks([HomeInteractorDelegate, SurveyRepository, AuthRepository])
+void main() {
+  describe("a Home interactor", () {
+    late HomeInteractor interactor;
+    late MockHomeInteractorDelegate delegate;
+    late MockSurveyRepository surveyRepository;
+    late MockAuthRepository authRepository;
+    late BehaviorSubjectGenerator generator;
+
+    beforeEach(() {
+      generator = BehaviorSubjectGenerator();
+
+      delegate = MockHomeInteractorDelegate();
+      when(delegate.authenticatedUserDidGet)
+          .thenAnswer((realInvocation) => generator.make(1));
+      when(delegate.surveysDidFetchFromCached)
+          .thenAnswer((realInvocation) => generator.make(2));
+      when(delegate.surveysDidFailToFetchFromCached)
+          .thenAnswer((realInvocation) => generator.make(3));
+      when(delegate.surveysDidFetchFromRemote)
+          .thenAnswer((realInvocation) => generator.make(4));
+      when(delegate.surveysDidFailToFetchFromRemote)
+          .thenAnswer((realInvocation) => generator.make(5));
+
+      surveyRepository = MockSurveyRepository();
+      locator.registerSingleton<SurveyRepository>(surveyRepository);
+
+      authRepository = MockAuthRepository();
+      locator.registerSingleton<AuthRepository>(authRepository);
+
+      interactor = HomeInteractorImpl();
+      interactor.delegate = delegate;
+    });
+
+    describe("it's getAuthenticatedUser is called", () {
```

Comment:
grammar issue: `its` not `it's` if you are talking about the getAuthenticatedUser method of the Interactor. Same for the rest.
Otherwise maybe just `getAuthenticatedUser is called` is enough.